ui/src/app/pages/data/data-set/dataset-page.tsx

```diff
@@ -856,398 +795,410 @@ export const DatasetPage = fp.flow(
+      return getIdsAndDomainsFromConceptSets(
+        d.conceptSets,
+        apiEnumToPrePackageConceptSets(d.prePackagedConceptSet)
+      );
+    };
+
+    const loadDataset = (dataset: DataSet, initialLoad?: boolean) => {
+      // This is to set the URL to reference the newly created dataset and the user is staying on the dataset page
+      // A bit of a hack but I couldn't find another way to change the URL without triggering a reload
+      if (window.location.href.endsWith('data-sets')) {
+        history.pushState({}, '', `${window.location.href}/${dataset.id}`);
       }
 
-      // If any domains were dropped, we want to drop any domain/value pair selections.
-      const droppedDomains = Array.from(
-        prevState.selectedDomainsWithConceptSetIds
-      ).filter((d) => !this.state.selectedDomainsWithConceptSetIds.has(d));
-      if (droppedDomains.length > 0) {
-        const updatedDomains = Array.from(
-          this.state.selectedDomainsWithConceptSetIds
-        ).map(({ domain }) => domain);
-        this.setState(
-          ({ selectedDomainsWithConceptSetIds, selectedDomainValuePairs }) => ({
-            selectedDomains: new Set(updatedDomains),
-            selectedDomainValuePairs: selectedDomainValuePairs.filter((p) =>
-              Array.from(selectedDomainsWithConceptSetIds).some(
-                (d) => d.domain === p.domain
-              )
-            ),
-          })
+      setDataSet(dataset);
+      setDataSetTouched(false);
+      // We only need to set selections on the initial load of a saved dataset,
+      // not for creating/updating since the selections will already be set
+      if (initialLoad) {
+        setIncludesAllParticipants(dataset.includesAllParticipants);
+        setSelectedConceptSetIds(dataset.conceptSets.map((cs) => cs.id));
+        setSelectedCohortIds(dataset.cohorts.map((c) => c.id));
+        setSelectedDomainValuePairs(
+          domainValuePairsToLowercase(dataset.domainValuePairs)
+        );
+        setSelectedDomains(getDomainsFromDataSet(dataset));
+        setSelectedDomainsWithConceptSetIds(
+          getDomainsWithConceptSetIdsFromDataSet(dataset)
+        );
+        setSelectedPrepackagedConceptSets(
+          apiEnumToPrePackageConceptSets(dataset.prePackagedConceptSet)
         );
       }
+    };
 
-      // After a state update, first check whether any new domains have been added.
-      const newDomains = Array.from(
-        this.state.selectedDomainsWithConceptSetIds
-      ).filter((d) => !prevState.selectedDomainsWithConceptSetIds.has(d));
-      if (!newDomains.length) {
-        return;
+    const fetchDataset = async () => {
+      const dataset = await dataSetApi().getDataSet(
+        workspace.namespace,
+        workspace.id,
+        +dataSetId
+      );
+      loadDataset(dataset, true);
+    };
+
+    useEffect(() => {
+      hideSpinner();
+      updatePrepackagedDomains();
+
+      (async () => await loadResources())();
```

Comment:
This should be equivalent to just calling `loadResources()`, and not waiting on the return promise. Maybe with a comment explaining that you're letting it run in the background